crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+
+  # (x)or any of these
+
+  # Cascading (function requires module requires package)
+  package: SuiAddress
+  module: String
+  function: String
+
+  kind: TransactionBlockKindInput
+  checkpoint: U64
+  fromAddress: SuiAddress
+  toAddress: SuiAddress
+
+  inputObject: SuiAddress
+  changedObject: SuiAddress
+}
+
+type ConsistentRange {
+  first: Checkpoint
+  last: Checkpoint
+}
+
+interface Owner {
+  id: SuiAddress!
+
+  objects(query: ObjectQuery): ObjectQueryConnection
+
+  balance(type: TypeInst!): Balance
+  balances: BalanceConnection
+
+  coins(type: TypeInst): CoinConnection
+
+  stakes: StakeConnection
+
+  defaultNameServiceName: String
+  nameServiceNames: NameServiceNamesConnection
+}
+
+# Returned by Object.owner, where we can't disambiguate between
+# Address and Object.
+type AmbiguousOwner implements Owner {}
+
+type Address implements Owner {
+  transactionBlocks(
+    filter: AddressTransactionBlockFilter,
+  ): TransactionBlockConnection
+}
+
+enum AddressTransactionBlockFilter {
+  SIGN # Transactions this address has signed
+  SENT # Transactions that transferred objects from this address
+  RECV # Transactions that received objects into this address
+  PAID # Transactions that were paid for by this address
+}
+
+type Object implements Owner {
+  owner: AmbiguousOwner
+  version: U64
+  digest: Digest
+
+  previousVersion: U64
+  previousTransactionBlock: TransactionBlock
+  storageRebate: U64
+
+  display: [DisplayEntry]
+
+  asMoveObject: MoveObject
+  asMovePackage: MovePackage
+
+  # Transaction Blocks that sent objects to this object
+  receivedTransactionBlocks: TransactionBlockConnection
+
+  dynamicField(dynamicFieldName: DynamicFieldName!): DynamicField
+  dynamicFields: DynamicFieldConnection
+
+  bcs: Bytes
+}
+
+type DisplayEntry {
+  key: String
+  value: String
+}
+
+type Epoch {
+  epochId: Int!
+  systemStateVersion: Int
+  protocolConfig: ProtocolConfig
+  referenceGasPrice: U64
+
+  systemParameters: SystemParameters
+  stakeSubsidy: StakeSubsidy
+  validatorSet: ValidatorSet
+  storageFund: StorageFund
+  safeMode: SafeMode
+
+  startTimestampMs: U64
+  endTimestampMs: U64
+
+  # TODO: Identify non-duplicate fields in `EndOfEpochInfo`
+
+  checkpoints: CheckpointConnection
+  transactionBlocks: TransactionBlockConnection
+}
+
+type ProtocolConfig {
+  protocolVersion: Int!
+  # Omitting for brevity
+}
+
+type SystemParameters {
+  durationMs: U64
+  stakeSubsidyStartEpoch: Int
+
+  minValidatorCount: Int
+  maxValidatorCount: Int
+
+  minValidatorJoiningStake: U64
+  validatorLowStakeThreshold: U64
+  validatorVeryLowStakeThreshold: U64
+  validatorLowStakeGracePeriod: Int
+}
+
+type StakeSubsidy {
+  balance: U64
+  distributionCounter: U64
+  currentDistributionAmount: U64
+  periodLength: Int
+  decreaseRate: Int
+}
+
+type ValidatorSet {
+  totalStake: U64
+
+  activeValidators: [Validator]
+
+  # Indices into `activeValidators`
+  pendingRemovals: [U64]
+
+  pendingActiveValidators: MoveObject
```

Comment:
The GraphQL translation of `ValidatorSet` replaces `ObjectID`s with a richer `MoveObject` type, to allow querying these objects' contents inline.